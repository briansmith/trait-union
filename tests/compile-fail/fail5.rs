use trait_union::trait_union;

trait F { }

trait_union! {
    union U<T>: F where T: Copy+'static = T;
}

fn main() { }
