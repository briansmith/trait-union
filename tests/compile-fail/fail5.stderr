error[E0277]: the trait bound `T: F` is not satisfied
 --> $DIR/fail5.rs:6:11
  |
6 |     union U<T>: F where T: Copy+'static = T;
  |           ^^^^  - required by this bound in `UVariant`
  |           |
  |           the trait `F` is not implemented for `T`
  |
help: consider further restricting this bound
  |
6 |     union U<T>: F where T: Copy+'static + F = T;
  |                                         ^^^
